<launch>

    <!-- 启动nezha_base底盘 -->
    <master auto="start"/>
    <node name="test_nezha_base" pkg="test_nezha_base" type="test_nezha_base_node" output="screen" respawn="true">
        <!--rosparam file="$(find test_nezha_base)/config/nezha_base.yaml" command="load"/ -->
	<param name="publish_odom_transform" value="true"/>
    <param name="control_rate" value="10"/>
    <param name="max_linear_speed" value="0.5"/> <!-- 最大速度 2m/s -->
    </node>
    <!-- 启动红外节点 -->
    < node name="infrared_driver" pkg="infrared_driver" type="infrared_driver" output="screen"/>

    <!-- 启动imu校准节点 -->
    < node name="apply_calib" pkg="imu_calib" type="apply_calib" output="screen" respawn="true">
	<remap from="raw" to="/imu_raw"/>
	<remap from="corrected" to="/imu"/>
	<param name="calib_file" value="$(find test_nezha_base)/config/imu_calib.yaml"/>
	<param name="calibrate_gyros" value="true"/>
	<param name="load_acc_" value="true"/>
	<param name="gyro_calib_samples" value="100"/>
    </node  >
    <node pkg="tf" type="static_transform_publisher" name="base_footprint_to_imu" args="-0.05 0 0.05 -1.57 0 0 base_footprint base_imu_link 100"/>
</launch>
